core/templates/pages/contributor-dashboard-page/services/contribution-and-review-backend-api.service.ts

```diff
@@ -0,0 +1,119 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Backend api service for fetching and resolving suggestions.
+ */
+
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { FetchSuggestionsResponse } from './contribution-and-review.service';
+import { UrlInterpolationService } from 'domain/utilities/url-interpolation.service';
+
+interface ResolveToExplorationData {
+  action: string;
+  'review_message': string;
+  'commit_message': string;
+}
+
+interface ResolveToSkillData {
+  action: string;
+  'review_message': string;
+  'skill_difficulty': string;
+}
+
+interface UpdateTranslationData {
+  'translation_html': string;
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ContributionAndReviewBackendApiService {
+  private _SUBMITTED_SUGGESTION_LIST_HANDLER_URL = (
+    '/getsubmittedsuggestions/<target_type>/<suggestion_type>');
+  private _REVIEWABLE_SUGGESTIONS_HANDLER_URL = (
+    '/getreviewablesuggestions/<target_type>/<suggestion_type>');
+  private _SUGGESTION_TO_EXPLORATION_ACTION_HANDLER_URL = (
+    '/suggestionactionhandler/exploration/<exp_id>/<suggestion_id>');
+  private _SUGGESTION_TO_SKILL_ACTION_HANDLER_URL = (
+    '/suggestionactionhandler/skill/<skill_id>/<suggestion_id>');
+  private _UPDATE_TRANSLATION_HANDLER_URL = (
+    '/updatetranslationsuggestionhandler/<suggestion_id>');
+  private _UPDATE_QUESTION_HANDLER_URL = (
+    '/updatequestionsuggestionhandler/<suggestion_id>');
+
+  constructor(
+    private http: HttpClient,
+    private urlInterpolationService: UrlInterpolationService
+  ) {}
+
+  async fetchSuggestionsAsync(
+      url: string, targetType: string, suggestionType: string
+  ): Promise<FetchSuggestionsResponse> {
+    let interpolatedUrl = this.urlInterpolationService.interpolateUrl(
+      this[url], {
```

Comment:
`this[url]` seems like a mistake. Is this intentional?

Reply:
This is intentional because this function receives different urls in different use cases

Reply:
What does url contain and how that url is a property of this?

Reply:
in `contribution-and-review-service`, there are four functions calling `_fetchSuggestionsAsync` with different urls (eg - `_REVIEWABLE_SUGGESTIONS_HANDLER_URL`, `_SUBMITTED_SUGGESTION_LIST_HANDLER_URL`) and `_fetchSuggestionsAsync` calls the function `fetchSuggestionsAsync` of the backend api service. That's why I used this[url] here and it refers to the urls such as `_REVIEWABLE_SUGGESTIONS_HANDLER_URL` and `_SUBMITTED_SUGGESTION_LIST_HANDLER_URL` from the backend api service. This was not a problem with the other functions of the backend api service because they were called using one url only and hence that url was referred from the inner function itself.

Reply:
I can change this approach (`this[url]`) if you want but that will increase the lines of code.

Reply:
@soumyo123-prog  @ashutoshc8101 Is this comment resolved?